jack.py

```diff
@@ -600,6 +680,14 @@ def transport_locate(self, frame):
         _check(_lib.jack_transport_locate(self._ptr, frame),
                "Error locating JACK transport")
 
+    def transport_query(self):
+        """return current transport position."""
+        # Test rolling transport status
+        if _lib.jack_transport_query(self._ptr, self.position) == 1:
+            return self.position
+        else:
+            return None
```

Comment:
This is my test script :


#!/usr/bin/env python3

"""JACK client that prints all received MIDI events."""

import jack
import binascii

client = jack.Client("MIDI-Monitor")
port = client.midi_inports.register("input")


def callback(frames, userdata):
     for offset, data in port.incoming_midi_events():
         # TODO: use ringbuffer
         print("{0}: 0x{1}".format(client.last_frame_time + offset,
binascii.hexlify(data).decode()))
         print("Time : {}".format(client.frames_since_cycle_start))
         position = client.transport_query()
         print(position.transport_info)
     return jack.CALL_AGAIN

client.set_process_callback(callback)

with client:
     print("#" * 80)
     print("press Return to quit")
     print("#" * 80)
     input()



Le 14/04/2015 16:43, Matthias Geier a écrit :
>
> In jack.py 
> <https://github.com/spatialaudio/jackclient-python/pull/10#discussion_r28332847>:
>
> > @@ -600,6 +680,14 @@ def transport_locate(self, frame):
> >          _check(_lib.jack_transport_locate(self._ptr, frame),
> >                 "Error locating JACK transport")
> >
> > +    def transport_query(self):
> > +        """return current transport position."""
> > +        # Test rolling transport status
> > +        if _lib.jack_transport_query(self._ptr, self.position) == 1:
> > +            return self.position
> > +        else:
> > +            return None
>
>     Yes I think better is to return a tuple ?
>
> Yes, I think returning a tuple would be good.
>
> The question is if we can improve on the numbers in the enum 
> |jack_transport_state_t|.
> We can probably wrap this into a custom type that displays something 
> like |jack.ROLLING| ...
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/spatialaudio/jackclient-python/pull/10/files#r28332847>.
>